src/functions/unlock.js

```diff
@@ -89,7 +89,15 @@ export async function unlock(
     }
   } catch (error) {
     // The the error caught was a 422 - Reference does not exist, this is OK - It means the lock branch does not exist
-    if (error.status === 422 && error.message === 'Reference does not exist') {
+
+    // debug the error msg
+    core.debug(`unlock() error.status: ${error.status}`)
+    core.debug(`unlock() error.message: ${error.message}`)
+
+    if (
+      error.status === 422 &&
+      error.message.startsWith('Reference does not exist')
```

Comment:
This is good, I would like to see a link referencing where this message is coming from, if the doc has it :)

Reply:
The docs only brief details around `422` status codes for when a Reference does not exist - [docs](https://docs.github.com/en/rest/git/refs?apiVersion=2022-11-28#delete-a-reference)

> - `422` Validation failed, or the endpoint has been spammed.

However, when you try to delete a Reference that doesn't exist (with the JS flavor of Octokit), you get back a `NotFoundError` with the following message: `Reference does not exist - https://docs.github.com/rest/git/refs#delete-a-reference`